### PR TITLE
Fix generate issue in finetuning script when finetuning with small datasets

### PR DESCRIPTION
**Pull request by rasbt**
We had an issue with the finetuning script hitting

https://github.com/Lightning-AI/litgpt/blob/6bf38fbcffa471bb5fe90f1d6afb544faf58bb0d/litgpt/generate/base.py#L123

during prompt evaluation when the instruction dataset training examples were small (e.g. 101). This PR fixes this issue. (It shouldn't impact the overall model, it's only fixing it for the prompt evaluation printed during training).

### CODE REVIEW

**Comment by rasbt** [comment]
Slight change of plans. Skipping the generation now with a warning if it's not possible. The generation during training is something some users don't want anyway. We can think about other options in the future (e.g., selecting an example from the validation set), but for now, this is a quick small fix I need when demoing LitGPT in a few days.
